src/XMakeBuildEngine/BackEnd/BuildManager/BuildParameters.cs

```diff
@@ -1002,7 +1005,11 @@ private void FindMSBuildExe()
             }
 
             // Use the default location of the directory from which the engine was loaded.
+#if FEATURE_APPDOMAIN
             path = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "MSBuild.exe");
+#else
+            path = Path.GetDirectoryName(Process.GetCurrentProcess().MainModule.FileName);
```

Comment:
> Is it safe to just use this all the time? That'd be easier.

@rainersigwald I'm not sure.  In general I'm wary of changing things for the .NET Framework build unless I'm confident they won't cause breaking changes if we merge them back into the next release of MSBuild for .NET Framework.